### PR TITLE
[FLINK-3331] TupleSerializer.getLength() can return fixed-length size

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/TupleSerializerBase.java [hunk 1]
```diff
@@ -56,7 +58,19 @@ public boolean isImmutableType() {
 
 	@Override
 	public int getLength() {
-		return -1;
+		if (length == -2) {
+			int sum = 0;
+			for (TypeSerializer<Object> serializer : fieldSerializers) {
+				if (serializer.getLength() != -1) {
```
Comment (StephanEwen) [inline flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/TupleSerializerBase.java]: How about making this check `serializer.getLength() > 0`? That way, we catch cases of "misbehaved" subclasses.

**Comment by StephanEwen** [comment]
Aside from a minor comment, this looks good.

File: flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/TupleSerializerBase.java (hunk 1 above)
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/TupleSerializerBase.java]: good idea, done.

**Comment by StephanEwen** [comment]
Looks good, +1 to merge this

**Comment by zentol** [comment]
Merging this.

**Pull request closed without merging**